packages/network-controller/tests/provider-api-tests/shared-tests.ts

```diff
@@ -0,0 +1,391 @@
+          describe(`method name: ${name}`, () => {
+            testsForRpcMethodSupportingBlockParam(name, {
+              providerType,
+              blockParamIndex,
+              numberOfParameters,
+            });
+          });
+        },
+      );
+    });
+
+    describe('other methods', () => {
+      describe('eth_getTransactionByHash', () => {
+        it("refreshes the block tracker's current block if it is less than the block number that comes back in the response", async () => {
+          const method = 'eth_getTransactionByHash';
+
+          await withMockedCommunications({ providerType }, async (comms) => {
+            const request = { method };
+
+            comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+            // This is our request.
+            comms.mockRpcCall({
+              request,
+              response: {
+                result: {
+                  blockNumber: '0x200',
+                },
+              },
+            });
+            comms.mockNextBlockTrackerRequest({ blockNumber: '0x300' });
+
+            await withNetworkClient(
+              { providerType },
+              async ({ makeRpcCall, clock, blockTracker }) => {
+                await makeRpcCall(request);
+                await waitForNextBlockTracker(blockTracker, clock);
+                expect(blockTracker.getCurrentBlock()).toStrictEqual('0x300');
+              },
+            );
+          });
+        });
+      });
+
+      describe('eth_getTransactionReceipt', () => {
+        it("refreshes the block tracker's current block if it is less than the block number that comes back in the response", async () => {
+          const method = 'eth_getTransactionReceipt';
+
+          await withMockedCommunications({ providerType }, async (comms) => {
+            const request = { method };
+
+            comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+            // This is our request.
+            comms.mockRpcCall({
+              request,
+              response: {
+                result: {
+                  blockNumber: '0x200',
+                },
+              },
+            });
+            comms.mockNextBlockTrackerRequest({ blockNumber: '0x300' });
+
+            await withNetworkClient(
+              { providerType },
+              async ({ makeRpcCall, blockTracker, clock }) => {
+                await makeRpcCall(request);
+                await waitForNextBlockTracker(blockTracker, clock);
+                expect(blockTracker.getCurrentBlock()).toStrictEqual('0x300');
+              },
+            );
+          });
+        });
+      });
+
+      describe('eth_chainId', () => {
+        it('does not hit the RPC endpoint, instead returning the configured chain id', async () => {
+          await withMockedCommunications({ providerType }, async (comms) => {
+            const request = { method: 'eth_chainId' };
+            comms.mockRpcCall({
```

Comment:
Something for a future PR: If this truly does not hit the RPC endpoint, then we shouldn't need this mock. If we don't, we should rename the test to match and also change the chain id to something else like '0x9999', otherwise it could be a false positive if the test is wrong and the code ends up automatically matches the currently selected Infura network.